homeassistant/components/xiaomi_miio/switch.py

```diff
@@ -881,8 +877,8 @@ async def async_update(self):
             if self._device_features & FEATURE_SET_POWER_MODE == 1 and state.mode:
                 self._state_attrs[ATTR_POWER_MODE] = state.mode.value
 
-            if self._device_features & FEATURE_SET_WIFI_LED == 1 and state.wifi_led:
-                self._state_attrs[ATTR_WIFI_LED] = state.wifi_led
+            if self._device_features & FEATURE_SET_LED == 1 and state.led:
+                self._state_attrs[ATTR_LED] = state.led
```

Comment:
An LED shouldn't be an attribute and have their own services for control. It should just be a light entity with entity category "config".

Reply:
Yes. Sonos does something similar with status light:

![image](https://user-images.githubusercontent.com/1444314/159981528-ce2d860f-a300-4bfb-97b4-e602ee5329fc.png)

Reply:
Agreed, I created this PR just to fix the deprecated warning so I will close this now. The real solution is to improve the backend library to allow creating entities based on what is supported by the given device, and remove all custom services altogether.

I put some thoughts on this topic in https://github.com/home-assistant/core/pull/67058#issuecomment-1057467676 but I'm afraid I won't personally have any free time in the future to pursue this.